### PR TITLE
API changes for molecular Consequences

### PR DESCRIPTION
**Pull request by Arraxx**
## Description

Added API changes for molecular Consequences and added find-subject-molecular-consequences in herokuapp.

### CODE REVIEW

**Comment by Arraxx** [comment]
please review @mihaitodor @rhdolin

File: app/common.py [hunk 1]
```diff
@@ -1817,3 +1888,83 @@ def query_transcript(transcript):
...
+
+def query_molecular_consequences_by_variants(normalized_variant_list, feature_consequence_list, query):
+    variant_list = []
+    for item in normalized_variant_list:
+        if "GRCh37" in item:
+            variant_list.append(item["GRCh37"])
+        if "GRCh38" in item:
+            variant_list.append(item["GRCh38"])
+
+    pipeline_part = [{'$match': {'$expr': {'$and': [{'$or': [{'$eq': ['$variantID', '$$myvariant_id']}]}]}}},
```
Comment (mihaitodor) [inline app/common.py:1901]: What is `$$myvariant_id`?

File: app/common.py [hunk 2]
```diff
@@ -1817,3 +1888,83 @@ def query_transcript(transcript):
...
+                    {'$lookup': {'from': 'MolecConseq', 'let': {'myvariant_id': '$variantID'}, 'pipeline': pipeline_part,
+                                 'as': 'molecularConsequenceMatches'}},
+                    {'$addFields': {}},
+                    {'$match': {'molecularConsequenceMatches': {'$exists': True, '$not': {'$size': 0}}}}]
+
+    try:
+        results = variants_db.aggregate(query_string)
+        results = list(results)
+    except Exception as e:
+        print(f"{e}")
```
Comment (mihaitodor) [inline app/common.py:1961]: Please use the same `print()` pattern that's already used throughout this file (smth like `print(f"DEBUG: Error{e} under query_molecular_consequences_by_variants query={query}")`)

File: app/common.py [hunk 3]
```diff
@@ -1817,3 +1888,83 @@ def query_transcript(transcript):
...
+        results = []
+
+    query_results = []
+
+    for item in results:
+        item["UUID"] = str(uuid4())
```
Comment (mihaitodor) [inline app/common.py:1967]: Why generate a new random UUID?

File: app/endpoints.py [hunk 4]
```diff
@@ -1199,6 +1199,113 @@ def find_subject_dx_implications(
...
+                    parameter = OrderedDict()
+                    parameter["name"] = "Molecular Consequence"
+                    parameter["part"] = []
+
+                    molecular_consequence_profile = common.create_molecular_consequence_profile(molecular_consequence, subject, [str(res['variantID'])])
+                    parameter["part"].append({
+                        "name": "Molecular Consequence",
+                        "resource": molecular_consequence_profile
+                    })
+                    # resource = common.create_fhir_variant_resource(
```
Comment (mihaitodor) [inline app/endpoints.py:1291]: Please remove commented-out code or add some comment to explain why it exists (same for line 1280)

**Review by mihaitodor — CHANGES_REQUESTED** [review]
Thanks Aman! I had a quick look and pointed out some potential issues.

Please also add some automated tests to https://github.com/FHIR/genomics-operations/tree/main/tests/integration_tests similar to the other operations.

File: app/common.py (hunk 3 above)
Comment (Arraxx) [reply app/common.py:1967]: Variant id should be unique.

File: app/common.py (hunk 1 above)
Comment (mihaitodor) [reply app/common.py:1901]: Oh, OK. I see. It's following an existing pattern (see `$$mygenotypeCode`).

**Comment by rhdolin** [comment]
this change is part of a new pull request

**Pull request closed without merging**